### PR TITLE
_latest Hive view & integrate with latest Metaflow

### PR DESCRIPTION
**Pull request by talebzeghmi**

### CODE REVIEW

File: datasets/plugins/batch/hive_dataset.py [hunk 1]
```diff
@@ -176,8 +179,13 @@ def write_spark(
                 tmp_partition_by += ["run_time"]
 
         df, partition_cols = self._write_data_frame_prep(df, partition_by=tmp_partition_by)
+
         _logger.info(f"write_spark({self.hive_table=}, {partition_cols=})")
+        _retry_with_backoff(partial(self._write_spark_helper, spark, df, partition_cols, **kwargs))
```
Comment (ashleybi-zg) [inline datasets/plugins/batch/hive_dataset.py:184]: do you want to add a comment or TODO about  the retry here so we remember why this is being used?

File: datasets/plugins/batch/hive_dataset.py [hunk 2]
```diff
@@ -203,8 +212,34 @@ def write_spark(
...
+    while True:
+        try:
+            return func()
+        except Exception as e:
+            if i >= retries:
+                print(f"Failed after {retries} retries:")
+                raise
+            else:
+                i += 1
+                sleep = backoff_in_seconds * 2 ** i + random.uniform(0, 5)
```
Comment (ashleybi-zg) [inline datasets/plugins/batch/hive_dataset.py:242]: this is to wait for tmp to be cleaned up?

File: datasets/plugins/batch/hive_dataset.py (hunk 1 above)
Comment (talebzeghmi) [reply datasets/plugins/batch/hive_dataset.py:186]: done!

File: datasets/plugins/batch/hive_dataset.py (hunk 2 above)
Comment (talebzeghmi) [reply datasets/plugins/batch/hive_dataset.py:244]: done!

**Pull request merged**